### ISSUE TITLE
question(query):  how to implement infinite scroll

### ISSUE THREAD

**Issue by gopinav** [prescottprue/redux-firestore#178]
Hi, this is more of a question regarding the best way to fetch data for virtual scroll. Right now, I am using the 'limit' query option. I use firestoreConnect to fetch INIT_LIMIT=10 documents on page load. When the user reaches the end of a list, I call a setListener with the same queryOptions but this time, limit set to INIT_LIMIT + 10. Reach end again, INIT__LIMIT + 20 and so on. Is this the correct way to do it? If I fetch 10 docs initially and then setListner for 20 docs, does it make 20 reads or just 10 additional reads? I want to understand this really well as I don't want to be re-fetching the same documents over and over again. I also have come across startAt(). But that would mean appending the next 10 documents to the existing list of documents. As I am directly mapping state to props with firestore reducer, I am not able to wrap my head around how to do it. 

**tl;dr - How to fetch additional n records on some event that can be triggered multiple times?**

**Comment by prescottprue** [comment]
Seems to the the same question as [630 of react-redux-firebase](https://github.com/prescottprue/react-redux-firebase/issues/630). As noted in there it may be useful to use something like [react-waypoint](https://github.com/brigade/react-waypoint) and use it to trigger [manual queries](http://react-redux-firebase.com/docs/queries.html#manual).
